src/git_handler.py

```diff
@@ -648,6 +649,36 @@ def find_open_pr_for_issue(issue_number: int) -> dict:
         log(f"Error searching for PRs related to issue #{issue_number}: {e}", is_error=True)
         return None
 
+def extract_issue_number_from_branch(branch_name: str) -> Optional[int]:
+    """
+    Extracts the GitHub issue number from a branch name with format 'copilot/fix-<issue_number>'.
+    
+    Args:
+        branch_name: The branch name to extract the issue number from
+        
+    Returns:
+        Optional[int]: The issue number if found and valid, None otherwise
+    """
+    if not branch_name:
+        return None
+    
+    # Use regex to match the exact pattern: copilot/fix-<number>
+    # This ensures we only match the expected format and extract just the number
+    pattern = r'^copilot/fix-(\d+)$'
+    match = re.match(pattern, branch_name)
+    
+    if match:
+        try:
+            issue_number = int(match.group(1))
+            # Validate that it's a positive number (GitHub issue numbers start from 1)
+            if issue_number > 0:
+                return issue_number
+        except ValueError:
+            # This shouldn't happen since \d+ only matches digits, but being safe
+            pass
+    
```

Comment:
could add a log message stating could not extract the issue number from branch name